migrateBuckets/migrateBuckets.js

```diff
@@ -0,0 +1,327 @@
+                        `users bucket: ${err}`);
+                } else {
+                    log.debug('Successfully deleted old bucket entry ' +
+                        'from users bucket');
+                }
+                return next();
+            }),
+        ], err => {
+            if (err) {
+                return cb(err);
+            }
+            log.debug('Successfully updated users..bucket entry');
+            return cb();
+        });
+    });
+}
+
+function _changeOneObjectAccount(bucket, object, log, cb) {
+    const {
+        bucketName,
+        destCanonicalId,
+        destOwnerName,
+    } = bucket;
+    const { Key, VersionId } = object;
+    const decodedId = VersionId ? versionIdUtils.decode(VersionId) : undefined;
+
+    // if versiondId is undefined, retrieval will return most current object md
+    const getOptions = { versionId: decodedId };
+
+    mdclient.getObject(bucketName, Key, log.getSerializedUids(),
+    (err, data) => {
+        const objectMD = JSON.parse(data);
+        if (err) {
+            log.error(`Error getting object metadata: ${err}`);
+            return cb(err);
+        }
+        const objectId = objectMD['owner-id'];
+        const objectOwner = objectMD['owner-display-name'];
+        if (objectId === destCanonicalId && objectOwner === destOwnerName) {
+            log.debug(`Object ${Key} id already updated`);
+            return cb();
+        }
+        objectMD['owner-id'] = destCanonicalId;
+        objectMD['owner-display-name'] = destOwnerName;
+        const putOptions = {};
+        if (VersionId) {
+            putOptions.versionId = VersionId;
+        }
+        return mdclient.putObject(bucketName, Key, JSON.stringify(objectMD),
+        log.getSerializedUids(), err => {
+            if (err) {
+                log.error(`Error updating object ${Key} metadata: ${err}`);
+                return cb(err);
+            }
+            log.debug(`Object ${Key} id successfully updated`);
+            return cb();
+        }, putOptions);
+    }, getOptions);
+}
+
+function _changeObjectsAccount(bucket, log, cb) {
+    const bucketName = bucket.bucketName;
+    const versioningEnabled = bucket.versioningEnabled;
+    const listMethod = versioningEnabled ? 'listObjectVersions' : 'listObjects';
+    const dataKey = versioningEnabled ? 'Versions' : 'Contents';
+    let moreObjects = true;
+    let objectsArr = [];
+    let marker = null;
+    async.whilst(
+        () => moreObjects === true,
+        done => {
+            const listParams = { Bucket: bucketName };
+            if (marker) {
+                listParams.Marker = marker;
+            }
+            s3[listMethod](listParams, (err, data) => {
+                if (err) {
+                    return done(err);
+                }
+                objectsArr = objectsArr.concat(data[dataKey]);
+                if (data.IsTruncated) {
+                    marker = data.Contents[data[dataKey].length - 1].Key;
```

Comment:
don't you need to use `data[dataKey]` instead of `data.Contents`? But see my above comment, with `listObjectVersions` it may not be necessary to distinguish treatment between non-versioned and versioned buckets anymore.